src/screens/CameraScreen.js

```diff
@@ -1,26 +1,296 @@
-import React from "react";
-import { View, Text, StyleSheet, Button } from "react-native";
-
-const CameraScreen = (props) => {
-  return (
-    <View style={styles.screen}>
-      <Text>Camera Screen</Text>
-      <Button
-        title="Posting"
-        onPress={() => {
-          props.navigation.navigate({ routeName: "Posting" });
-        }}
-      />
-    </View>
-  );
-};
+import React, { Component } from "react";
+import { View, Text, SafeAreaView, StatusBar, StyleSheet } from "react-native";
+import { MaterialIcons } from "@expo/vector-icons";
+import { Camera } from "expo-camera";
+import * as Permissions from "expo-permissions";
+import { Video } from "expo-av";
+import { TouchableOpacity } from "react-native-gesture-handler";
+import Colors from "../constants/Colors";
+
+export default class CameraScreen extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      hasCameraPermissions: null,
+      type: Camera.Constants.Type.back,
+      flash: Camera.Constants.FlashMode.off,
+      videoTaken: false,
+      cameraIsRecording: false,
+      video: null,
+    };
+  }
+
+  componentDidMount = async () => {
+    const camera = await Permissions.askAsync(Permissions.CAMERA);
+    this.setState({
+      hasCameraPermissions: camera.status === "granted",
+    });
+  };
+
+  _changeType = () => {
+    this.setState((prevState) => {
+      if (prevState.type === Camera.Constants.Type.back) {
+        return { type: Camera.Constants.Type.front };
+      } else {
+        return { type: Camera.Constants.Type.back };
+      }
+    });
+  };
+
+  _changeFlash = () => {
+    this.setState((prevState) => {
+      if (prevState.flash === Camera.Constants.FlashMode.off) {
+        return { flash: Camera.Constants.FlashMode.on };
+      } else if (prevState.flash === Camera.Constants.FlashMode.on) {
+        return { flash: Camera.Constants.FlashMode.auto };
+      } else if (prevState.flash === Camera.Constants.FlashMode.auto) {
+        return { flash: Camera.Constants.FlashMode.off };
```

Comment:
Have you tested your physical device for flash?

Reply:
Yes I am working on that part along add bar on top of the camera screen for timer.